### PR TITLE
Verbose `enable_if` in batched linear algebra

### PR DESCRIPTION
**Pull request by sslattery**
Add more verbose enable_if to batched linear algebra for CUDA deduction in attempts to overcome the CUDA build errors on Jenkins in #9

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -27,10 +27,10 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
 find_package(CLANG_FORMAT 10)
 find_package(MPI REQUIRED)
 find_package(Kokkos REQUIRED)
-find_package(Cabana REQUIRED COMPONENTS Cabana::Cajita Cabana::cabanacore)
 find_package(HYPRE REQUIRED)
-find_package(ArborX REQUIRED)
 find_package(Boost 1.66.0 REQUIRED)
+find_package(ArborX REQUIRED)
+find_package(Cabana REQUIRED COMPONENTS Cabana::Cajita Cabana::cabanacore)
```
Comment (dalg24) [inline CMakeLists.txt:33]: Did you mean to commit these changes?

File: src/Picasso_BatchedLinearAlgebra.hpp [hunk 2]
```diff
@@ -1586,40 +1586,52 @@ KOKKOS_INLINE_FUNCTION auto operator|( const ExpressionX& x,
 // Scalar multiplication.
 //---------------------------------------------------------------------------//
 // Matrix.
-template <class ExpressionA,
-          typename std::enable_if_t<is_matrix<ExpressionA>::value, int> = 0>
-KOKKOS_INLINE_FUNCTION auto
-operator*( const typename ExpressionA::value_type& s, const ExpressionA& a )
+template <class ExpressionA, class Scalar,
+          typename std::enable_if_t<
```
Comment (dalg24) [inline src/Picasso_BatchedLinearAlgebra.hpp:1590]:
```suggestion
          std::enable_if_t<
```

File: CMakeLists.txt (hunk 1 above)
Comment (sslattery) [reply CMakeLists.txt:33]: yes we were surprisingly getting configure errors with HYPRE if they weren't in this order.

Comment (streeve) [reply CMakeLists.txt:33]: I opened #40 - hopefully that works for you as well

**Comment by streeve** [comment]
Closing - this did not fix the issue

**Pull request closed without merging**